### DISCUSSION TITLE
regex to ensure only valid unicode letters

### DISCUSSION THREAD

**Discussion by jrnxf** [react-hook-form/react-hook-form#3146] [Show and tell]
I think it's rude to prevent someone from using their real name on your site by forcing them into the `[a-zA-Z]` restriction. The regex used here should accept any letter from any alphabet. Hopefully this can lead to some more inclusive designs. Cheers!

```
<Input
  label="Username"
  name="username"
  register={() =>
    _register({
      required: 'Username is a required field',
      validate: val =>
        val?.match(/\p{L}/gu)?.join('') === val || 'Username must contain only valid unicode letters',
    })
  }
  errors={errors}
/>
```

**Comment by bluebill1049** [comment]
I am confused, what's this about?

**Reply by jrnxf** [reply]
This is useful for accepting all letters from any alphabet into your form. Whereas `[a-zA-Z]` limits users to only using the standard english alphabet. I'm using this on an app I'm building to allow users the chance to user their real name.

**Reply by bluebill1049** [reply]
yeah, the lib offers you the ability to supply any regex as you want, we don't restrict any patterns.

**Reply by jrnxf** [reply]
I can tell you that this regex pattern `/\p{L}/gu` was not working using the pattern property. That's why I used the validate property. If you'd like I can raise an issue for it?

**Reply by CodeToTee** [reply]
I'm also seeing a problem with using 'pattern' to check with a RegEx.  Basically, it doesn't work (all cases pass).  If I use a custom, 'validation' function (like @coloradocolby did), and use the same, exact RegEx pattern, it validates properly.  I'm seeing this in the latest version (6.9.6)

**Reply by CodeToTee** [reply]
> I'm also seeing a problem with using 'pattern' to check with a RegEx. Basically, it doesn't work (all cases pass). If I use a custom, 'validation' function (like @coloradocolby did), and use the same, exact RegEx pattern, it validates properly. I'm seeing this in the latest version (6.9.6)

Please ignore above comment.  I had an issue with my render props function inside TextInput in a Controller...my bad.  Sorry for the churn!  BTW...this is a great library!

**Reply by bluebill1049** [reply]
awesome! glad to know it's working.
